src/solver/impls/arkode/arkode_mri.cxx

```diff
@@ -0,0 +1,990 @@
+#include "bout/msg_stack.hxx"
+#include "bout/options.hxx"
+#include "bout/output.hxx"
+#include "bout/unused.hxx"
+#include "bout/utils.hxx"
+
+#include <arkode/arkode_arkstep.h>
+#include <arkode/arkode_bbdpre.h>
+#include <sundials/sundials_math.h>
+#include <sundials/sundials_types.h>
+
+#include <algorithm>
+#include <numeric>
+
+class Field2D;
+
+// NOLINTBEGIN(readability-identifier-length)
+namespace {
+int arkode_rhs_s_explicit(BoutReal t, N_Vector u, N_Vector du, void* user_data);
+int arkode_rhs_s_implicit(BoutReal t, N_Vector u, N_Vector du, void* user_data);
+int arkode_rhs_f_explicit(BoutReal t, N_Vector u, N_Vector du, void* user_data);
+int arkode_rhs_f_implicit(BoutReal t, N_Vector u, N_Vector du, void* user_data);
+int arkode_s_rhs(BoutReal t, N_Vector u, N_Vector du, void* user_data);
+int arkode_f_rhs(BoutReal t, N_Vector u, N_Vector du, void* user_data);
+
+int arkode_s_bbd_rhs(sunindextype Nlocal, BoutReal t, N_Vector u, N_Vector du,
+                   void* user_data);
+int arkode_f_bbd_rhs(sunindextype Nlocal, BoutReal t, N_Vector u, N_Vector du,
+                   void* user_data);
+int arkode_s_pre(BoutReal t, N_Vector yy, N_Vector yp, N_Vector rvec, N_Vector zvec,
+               BoutReal gamma, BoutReal delta, int lr, void* user_data);
+int arkode_f_pre(BoutReal t, N_Vector yy, N_Vector yp, N_Vector rvec, N_Vector zvec,
+               BoutReal gamma, BoutReal delta, int lr, void* user_data);
+
+} // namespace
+// NOLINTEND(readability-identifier-length)
+
+ArkodeMRISolver::ArkodeMRISolver(Options* opts)
+    : Solver(opts), diagnose((*options)["diagnose"]
+                                 .doc("Print some additional diagnostics")
+                                 .withDefault(false)),
+      mxsteps((*options)["mxstep"]
+                  .doc("Maximum number of steps to take between outputs")
+                  .withDefault(500)),
+      treatment((*options)["treatment"]
+                    .doc("Use default capability (imex) or provide a specific treatment: "
+                         "implicit or explicit")
+                    .withDefault(MRI_Treatment::ImEx)),
+      inner_treatment((*options)["inner_treatment"]
+                    .doc("Use default capability (imex) or provide a specific inner_treatment: "
+                         "implicit or explicit")
+                    .withDefault(MRI_Treatment::ImEx)),
+      set_linear(
+          (*options)["set_linear"]
+              .doc("Use linear implicit solver (only evaluates jacobian inversion once)")
+              .withDefault(false)),
+      inner_set_linear(
+          (*options)["inner_set_linear"]
+              .doc("Use linear implicit solver (only evaluates jacobian inversion once)")
+              .withDefault(false)),
+      fixed_step((*options)["fixed_step"]
+                     .doc("Solve explicit portion in fixed timestep mode. NOTE: This is "
+                          "not recommended except for code comparison")
```

Comment:
This comment should be updated.  In the MRI context here, I believe that the option determines whether both the slow and fast solvers will use fixed step sizes (i.e., nothing to do with "explicit").  
```suggestion
                     .doc("Solve both fast and slow time scales using fixed time step sizes. NOTE: This is "
                          "not recommended except for code comparison")
```